1_fetch/src/data_summary.R

```diff
@@ -11,112 +11,62 @@
-  
-  dfstats <- data.frame("seg_id_nat" = sites$seg_id_nat[[1]], 
-                        "Month" = sites$month[[1]],
-                        "max_temp_observed" = max(sites$max_temp_c, na.rm = TRUE),
-                        "mean_monthly_temp" = mean(sites$mean_temp_c, na.rm = TRUE),
-                        "min_temp_observed" = min(sites$min_temp_c, na.rm = TRUE),
-                        "Date Range" = lubridate::as.interval(start = sites$date[[1]], sites$date[[nrow(sites)]]),
-                        #"Intercept" = stats[1], 
-                        "Slope" = stats[2] 
-                        # "Date Range" = descrip_site$date_range 
-                        # "Max Temp Observed" = descrip_site$max_temp_observed,
-                        # "Min Temp Observed" = descrip_site$min_temp_observed
-                        #"r-squared" = stats[3])
-                        )
-  return(dfstats)
-}
-
 flexible_linear_regression <- function(sites, type)
 {
   #where sites is each branched target, and type is the type of linear regression that is be run 
   if (type == 1)
   {
     lr <- lm(month_mean ~ year, data = sites)
     sum_lr <- summary(lr)
+    r_cor <- cor(year, month_mean) 
+    max_temp <-  max(sites$month_mean, na.rm = TRUE)
+    mean_temp <- mean(sites$month_mean, na.rm = TRUE)
+    min_temp <-  min(sites$month_mean, na.rm = TRUE)
   }
   else if (type == 2)
   {
     #2 will be the mean of max's. 
     lr <- lm(month_meanOfMax ~ year, data = sites)
     sum_lr <- summary(lr)
+    r_cor <- cor(year, month_meanOfMax) 
+    max_temp <-  max(sites$month_meanOfMax, na.rm = TRUE)
+    mean_temp <- mean(sites$month_meanOfMax, na.rm = TRUE)
+    min_temp <-  min(sites$month_meanOfMax, na.rm = TRUE)
   }
   else if (type == 3)
   {
     lr <- lm(month_meanOfMin ~ year, data = sites)
     sum_lr <- summary(lr)
+    r_cor <- cor(year, month_meanOfMin) 
+    max_temp <-  max(sites$month_meanOfMin, na.rm = TRUE)
+    mean_temp <- mean(sites$month_meanOfMin, na.rm = TRUE)
+    min_temp <-  min(sites$month_meanOfMin, na.rm = TRUE)
   }
   else if (type == 4)
   {
     lr <- lm(annual_mean ~ year, data = sites)
     sum_lr <- summary(lr)
+    
+    r_cor <- cor(sites$year, sites$annual_mean) 
+    max_temp <-  max(sites$annual_mean, na.rm = TRUE)
+    mean_temp <- mean(sites$annual_mean, na.rm = TRUE)
+    min_temp <-  min(sites$annual_mean, na.rm = TRUE)
   }
   stats <- c(sum_lr$coefficients[[1]],
              sum_lr$coefficients[[2]])
   
   dfstats <- data.frame("seg_id_nat" = sites$seg_id_nat[[1]], 
                         "Month" = sites$month[[1]],
-                        "max_temp_observed" = max(sites$max_temp_c, na.rm = TRUE),
-                        "mean_monthly_temp" = mean(sites$mean_temp_c, na.rm = TRUE),
-                        "min_temp_observed" = min(sites$min_temp_c, na.rm = TRUE),
+                        "max_temp_observed" = max_temp,
```

Comment:
This is the construction of the new data frame, I will upload a sample of it to the share point for you to inspect.

Reply:
Current output features are:
"seg_id_nat"        "Month"             "max_temp_observed" "mean_monthly_temp" "min_temp_observed" "Date.Range"  "years"             "Slope"             "r"                 "r2"